src/v/cluster/tm_stm_types.h

```diff
@@ -15,16 +15,63 @@
 namespace cluster {
 
 enum tx_status : int32_t {
-    ongoing,
-    preparing,
-    prepared,
-    aborting, // abort is initiated by a client
-    killed,   // abort is initiated by a timeout
-    ready,
-    tombstone,
+    /**
+     * When transactional id to producer id mapping is added to coordinator it
+     * starts in this state
+     */
+    empty = 5,
+    /**
+     * Transaction is ongoing as soon as partition or offset is added to it
+     */
+    ongoing = 0,
+    /**
+     * Transaction is marked as preparing_commit when it is requested to be
+     * committed by the client but before committing transaction on data and
+     * offset partitions
+     */
+    preparing_commit = 1,
+    /**
+     * Transaction is completed commit when its participants committed the
+     * transaction. This state is not replicated as committing data on
+     * participants is idempotent and can be retried by the new leader.
+     */
+    completed_commit = 2,
+    /**
+     * Transaction is marked as preparing_abort when it is requested to be
+     * aborted by the client but before committing transaction on data and
+     * offset partitions
+     */
+    preparing_abort = 3,
+    /**
+     * The same as the preparing_abort but when transaction is aborted by
+     * internals of Redpand f.e. timeout or transactional id limit overflow
+     */
+    expired = 4,
+    /**
+     * Transaction is completed commit when its participants aborted the
+     * transaction. This state is not replicated as aborting data on
+     * participants is idempotent and can be retried by the new leader.
+     */
+    completed_abort = 7,
+    /**
+     * Transaction is about to be removed
```

Comment:
nit: a little more detail here may be useful, under what context is this state possible.